### PR TITLE
Added pico w tutorial

### PR DESCRIPTION
**Pull request by malw4**
My changes are in the RaspberryPi-Pico-W file.
Implementing_LwM2M_objects_on_RaspberryPi file was only dragged to this folder.

### CODE REVIEW

**Comment by malw4** [comment]
@JZimnol review please

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+
+   ![Temperature object](images/raspi1.png "3303")
+
+## Step 5: Implement the LwM2M push button based on the Multiple Axis Joystick object `3345`
+
+Now you can implement the Push Button module based on the OMA DM Multiple Axis Joystick object (3345).
+
+0. Type in the terminal:
+      ```
+      $ sudo svetovid-fsdmtool generate --object 3345 --output-dir /etc/svetovid/dm --generator python`
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:248]:
```suggestion
      $ sudo svetovid-fsdmtool generate --object 3345 --output-dir /etc/svetovid/dm --generator python
```

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+      ```
+
+    !!! tip
+        You can analyze the resource implementation in the `/etc/svetovid/dm/3345` folder. For more details, see the [Svetovid page on GitHub](https://github.com/AVSystem/Svetovid-raspberry-client).
+
+0. Type in the terminal: `cd /etc/svetovid/dm/3345/resources ./5550 read`.
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:254]:
```suggestion
0. Type in the terminal: `cd /etc/svetovid/dm/3345/resources && ./5550 read`.
```

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+
+      if __name__ == '__main__':
+          ResourceHandler_3345_5501().main()
+      ```
+0. Plug in the push button to digital port D3 of the GrovePi/Sense HAT.
+0. Restart Svetovid:
+    `sudo systemctl restart svetovid.service --now`
+0. Refresh device state in Coiote and see if the Multiple Axis Joystick object `3345` has appeared in the Objects tab and values are reported for the Digital Input State `5500` and Digital Input Counter `5501` resources:
+
+       ![Joystick button object](images/button.png "3345")
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:474]: Missing image here.

File: docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+## Prerequisites
+
+- The Raspberry Pi Pico W board with a USB cable.
+- A user with access to the Coiote IoT Device Management platform and appropriate permissions.
+
+## Prepare binaries
+### Step 1: Create a workspace
+
+0. Create a project directory for the integration.
+0. Clone Raspberry Pi Pico SDK and FreeRTOS kernel into a base directory, parallel to Anjay-pico-client Github repository:
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md:14]: I would **bold** the `Raspberry Pi Pico SDK` and `FreeRTOS kernel` names.

File: docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+### Compile the application
+0. Go to the Anjay-pico-client repository and update submodules:
+    ```
+    cd Anjay-pico-client
+    git submodule update --init --recursive
+    mkdir build && cd build
+    ```
+
+0. Build the project with `<ssid>` and `<pass>` replaced with your WIFI name and password respectively. LwM2M Client Endpoint Name is also configured by `<endpoint_name>` parameter:
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md:38]: What about `<identity>` and `<psk>`?

File: docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+Then use `make -j` command to generate directories named after examples that contain files with .uf2 and .hex extensions.
+
+!!! note
+    If you're using Windows system you should add `-G "MinGW Makefiles"` at the end of the cmake command. It will help generate proper files.
+
+## Flash the board using bootloader
+
+0. To program using the bootloader, press BOOTSEL button while connecting Raspberry Pi Pico W through a USB cable - it should be recognized as a Mass Storage device.
+0. Copy the .uf2 file to the open Mass Storage device folder.
+0. After Pico will be programmed, reset and start running the code.
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md:56]:
```suggestion
0. One Pico is programmed, the board will restart and start running the code.
```
Or something like that.

File: docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+To connect the board:
+
+1. Log in to Coiote DM and from the left side menu, select **Device Inventory**.
+2. In **Device Inventory**, click **Add device**.
+3. Select the **Connect your LwM2M device directly via the Management server** tile.
+       ![Add via Mgmt](images/mgmt_tile.png "Add via Mgmt")
+    3. In the **Device credentials** step:
+         - In the **Device ID** enter your board endpoint name, e.g. `test_device`.
+             ![Device credentials step](images/add_mgmt_quick.png "Device credentials step")
+         - In the **Security mode** section, select the security mode you choose in [Compile the application](#compile-the-application) step:
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md:71]:
```suggestion
         - In the **Security mode** section, select the security mode you've choosen in [Compile the application](#compile-the-application) step.
```

File: docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+        !!! note
+            If you choose **Pre-Shared Key** mode then type:
+
+            - In the **Key identity** field, the same name as in the Endpoint name field.
+            - In the **Key** field, the shared secret used in the device-server authentication.
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/RaspberryPi-Pico-W.md:77]: Can you move this `Note` more to the right so it's fully below the `In the Security mode section` bullet point?

File: mkdocs.yml [hunk 9]
```diff
@@ -36,7 +36,9 @@ nav:
           - 'index.md'
         - 'Anjay integrations':
             - 'Anjay_integrations/Getting_started.md'
-            - 'Anjay_integrations/Implementing_LwM2M_objects_on_RaspberryPi.md'
+            - 'Raspberry Pi Pico boards':
```
Comment (JZimnol) [inline mkdocs.yml:39]:
```suggestion
            - 'Raspberry Pi boards':
```

**Review by JZimnol — COMMENTED** [review]
@malw4 reviewed. There are some nits and some important suggestions. 
I know that you've only moved the `Implementing LwM2M objects...` file, but I have some suggestions there too. Please look at them.

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 10]
```diff
@@ -0,0 +1,474 @@
+# Tutorial: Implementing LwM2M objects on RaspberryPi
+
+## Introduction
+
+Svetovid is a LwM2M client that gives you tools to implement selected LwM2M objects for [RaspberryPi-based](https://www.raspberrypi.org/) IoT devices. The Python programming language is used to implement LwM2M objects. Svetovid is based on a state-of-the-art LwM2M client Anjay developed by AVSystem.
+It features the FSDM (File System Data Model) in which LwM2M objects are mapped to folders that follow a particular schema, and executables behave as expected by the LwM2M client that loads and manages them.
```
[the remark this answers is not in the document]
Comment (malw4) [reply docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:6]: @mkrawczyk-avs what do you think?

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 11]
```diff
@@ -0,0 +1,490 @@
...
+ To connect to Coiote IoT Device Management LwM2M Server, please register at [https://eu.iot.avsystem.cloud](https://eu.iot.avsystem.cloud).
+
+ To connect the board:
+
+ 1. Log in to Coiote DM and from the left side menu, select **Device Inventory**.
+ 2. In **Device Inventory**, click **Add device**.
+ 3. Select the **Connect your LwM2M device directly via the Management server** tile.
+       ![Add via Mgmt](images/mgmt_tile.png "Add via Mgmt")
+    3. In the **Device credentials** step:
+         - In the **Device ID** enter your board endpoint name, e.g. `test_device`.
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:142]: `e.g. test_device` but you have `raspberry_pi_test` on the screenshot.

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 12]
```diff
@@ -0,0 +1,490 @@
...
+                - In the **Key identity** field, the same name as in the Endpoint name field.
+                - In the **Key** field, the shared secret used in the device-server authentication.
+
+    4. Click the **Add device** button and **Confirm** in the confirmation pop-up.
+
+ 0. Analyze the objects exposed by the device.
+    ![Device objects](images/rasp.png "Objects")
+
+!!! note
+    Your RaspberryPi-based device will feature a number of default LwM2M objects provided by Svetovid, for instance the temperature object `/3303`.
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:158]: There is no `/3303` object on the screenshot (it is added later in the article). Maybe change it to the `3311 Light Control`?

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 13]
```diff
@@ -0,0 +1,490 @@
...
+## Step 5: Implement the LwM2M temperature object `/3303` (only for GrovePi)
+
+0. Disable the Svetovid service:
+
+    `sudo systemctl disable svetovid.service --now`
+
+0. Generate a stub object in your `dm` (data model) Svetovid folder by pasting and executing the following command in the terminal: `$ sudo svetovid-fsdmtool generate --object 3303 --output-dir /etc/svetovid/dm --generator python`
+
+    !!! tip
+        You can analyze the resource implementation in the `/etc/svetovid/dm/3303 folder`. For more details, see the [Svetovid page on GitHub](https://github.com/AVSystem/Svetovid-raspberry-client).
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:169]:
```suggestion
        You can analyze the resource implementation in the `/etc/svetovid/dm/3303` folder. For more details, see the [Svetovid page on GitHub](https://github.com/AVSystem/Svetovid-raspberry-client).
```

**Review by JZimnol — COMMENTED** [review]
@malw4 second round of review completed.

**Review by laurensslats — CHANGES_REQUESTED** [review]

File: docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md [hunk 14]
```diff
@@ -0,0 +1,490 @@
...
+1. To install the GrovePi repository, paste and execute the following commands:
+
+    ```
+        mkdir ~/Dexter
+        cd ~/Dexter
+        git clone https://github.com/DexterInd/GrovePi
+        cd ~/Dexter/GrovePi/Script
+        bash ./update_grovepi.sh
+    ```
+2. Check if the GrovePi libraries are working correctly:
```
Comment (JZimnol) [inline docs/Anjay_integrations/RaspberryPi_integration/Implementing_LwM2M_objects_on_RaspberryPi.md:49]:
```suggestion
        bash ./update_grovepi.sh
    ```

2. Check if the GrovePi libraries are working correctly:
```

The "1." and "2." are still not in line :/

**Review by JZimnol — COMMENTED** [review]
@malw4 Added one last comment.

**Comment by JZimnol** [comment]
@malw4 approved.

**Pull request merged**